src/haddock/modules/analysis/caprieval/__init__.py

```diff
@@ -69,6 +73,92 @@ def is_nested(models: list[Union[PDBFile, list[PDBFile]]]) -> bool:
                 return True
         return False
 
+    def validate_reference(self, reference: Path) -> Path:
+        """Validate the reference file by returning only one model.
+
+        Parameters
+        ----------
+        reference : Path
+            Path to the input reference structure, possibly containing
+            an ensemble.
+
+        Returns
+        -------
+        reference or first_model_path : Path
+            Path to the reference structure to be used downstream.
+        """
+        # Extremly complicated stuff to manage the gathering of the sys.stdout,
+        # as the pdb_tools.pdb_wc is basically writing on it.
+        import sys
+        from io import TextIOWrapper, BytesIO
+        # Memorize previous sys.stdout
+        original_stdout = sys.stdout
+        # setup the new stdout environment
+        sys.stdout = TextIOWrapper(BytesIO(), sys.stdout.encoding)
+
+        # Count number of models
+        with open(reference, "r") as fh:
+            pdb_wc(fh, "m")
+        # Get output
+        sys.stdout.seek(0)  # Jump to the start
+        wc_return = sys.stdout.read()  # Read output
+        # Restore original stdout
+        sys.stdout.close()
+        sys.stdout = original_stdout
+        # Parse output
+        for line in wc_return.split("\n"):
+            if "No. models" in line:
+                sline = line.strip().split()
+                nb_models = int(sline[-1])
+                break
```

Comment:
extremely complex indeed...

Reply:
yes, the pdb_tools `pdb_wc` is not returning anything... :'(